tiledb/sm/storage_manager/storage_manager.cc

```diff
@@ -1751,54 +1748,59 @@ Status StorageManager::load_array_schema(
 Status StorageManager::load_fragment_metadata(
     OpenArray* open_array,
     const EncryptionKey& encryption_key,
-    bool* in_cache) {
+    bool* in_cache,
+    uint64_t timestamp) {
   // Get all the fragment uris, sorted by timestamp
   std::vector<URI> fragment_uris;
   const URI& array_uri = open_array->array_uri();
   RETURN_NOT_OK(get_fragment_uris(array_uri, &fragment_uris));
-  sort_fragment_uris(&fragment_uris);
 
+  // Check if the array is empty
   if (fragment_uris.empty())
     return Status::Ok();
 
   *in_cache = false;
+  // Sort the URIs by timestamp
+  std::vector<std::pair<uint64_t, URI>> sorted_fragment_uris;
+  sort_fragment_uris(fragment_uris, &sorted_fragment_uris);
 
   // Load the metadata for each fragment, only if they are not already loaded
-  std::vector<FragmentMetadata*> fragment_metadata;
-  for (auto& uri : fragment_uris) {
-    auto metadata = open_array->fragment_metadata_get(uri);
-    if (metadata == nullptr) {
+  auto latest_timestamp = open_array->latest_timestamp();
+  for (auto& sf : sorted_fragment_uris) {
+    auto frag_timestamp = sf.first;
+    auto uri = sf.second;
+    if (frag_timestamp > latest_timestamp && frag_timestamp <= timestamp) {
```

Comment:
Why the extra requirement of `frag_timestamp > latest_timestamp`?

Reply:
This function loads only the fragments that have been not loaded yet. This condition ensures that the function does not reload an already loaded fragment.